pkg/resource/notebook_instance/hooks.go

```diff
@@ -48,3 +52,33 @@ func (rm *resourceManager) customSetOutput(r *resource) {
 	latestStatus := r.ko.Status.NotebookInstanceStatus
 	svccommon.SetSyncedCondition(r, latestStatus, &resourceName, &modifyingStatuses)
 }
+
+// CustomSetOutputDescribe has two functions
+// 1. Setting the resource synced condition during sdk.Find
+// 2. Starting the Notebook if the StoppedByController field is set to UpdateTriggerd
+// 3. Reseting the StoppedByController field.
+func (rm *resourceManager) customSetOutputDescribe(r *resource,
+	ko *svcapitypes.NotebookInstance) error {
+	notebookStatus := *ko.Status.NotebookInstanceStatus // Get the Notebook State
+	if notebookStatus == svcsdk.NotebookInstanceStatusStopped {
+		if ko.Status.StoppedByControllerMETA != nil && *ko.Status.StoppedByControllerMETA == "UpdateTriggered" {
+			err := rm.startNotebookInstance(r)
+			if err != nil {
+				//We dont update StoppedByController here because we want it to try to start it again on the next reqeue.
+				return err
+			}
+			ko.Status.StoppedByControllerMETA = nil
+		}
+	}
+	rm.customSetOutput(&resource{ko}) // We set the sync status here
+	return nil
+}
+
+// customSetOutputUpdate sets the StoppedByController field to UpdateTriggered if it was in UpdatePending before.
+func (rm *resourceManager) customSetOutputUpdate(ko *svcapitypes.NotebookInstance, latest *resource) {
+	//TODO: Replace the IsUpdating status with an annotation if the runtime can update annotations after a readOne call.
```

Comment:
`isUpdating` is not in use anymore. However, the new field we added could also be converted into an annotation later.